Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1918,66 +1937,134 @@ void CGContextSetPatternPhase(CGContextRef context, CGSize phase) {
-        !transformByGlyph) {
-        RETURN_IF_FAILED(
-            Draw(_kCGCoordinateModeUserSpace, &textTransform, [&state, glyphRun](CGContextRef context, ID2D1DeviceContext* deviceContext) {
-                deviceContext->DrawGlyphRun(D2D1::Point2F(0, 0), glyphRun, state.fillBrush.Get(), DWRITE_MEASURING_MODE_NATURAL);
-                return S_OK;
-            }));
-    } else {
-        // Using device space here gives us finer-grained control over the transform.
+        // Transform the geometry by the final transformation to be stroked/clipped correctly
         RETURN_IF_FAILED(
-            Draw(_kCGCoordinateModeDeviceSpace,
-                 nullptr,
-                 [&state, &glyphRun, &deviceTransform, &textTransform](CGContextRef context, ID2D1DeviceContext* deviceContext) {
-                     CGAffineTransform runningGlobalTransform = deviceTransform;
-                     // Text scaling and rotation apply to each glyph relative to its origin, so we must draw each glyph
-                     // transformed
-                     // independently
-                     DWRITE_GLYPH_RUN individualGlyphRun{ glyphRun->fontFace,
-                                                          glyphRun->fontEmSize,
-                                                          1, // Since this is glyph by glyph, glyphCount is one
-                                                          glyphRun->glyphIndices,
-                                                          nullptr,
-                                                          nullptr,
-                                                          glyphRun->isSideways,
-                                                          glyphRun->bidiLevel };
-                     D2D1_POINT_2F origin{ 0, 0 };
-                     // Iterate through every glyph by incrementing pointer in glyphIndices array
-                     for (uint32_t i = 0; i < glyphRun->glyphCount; ++i, ++(individualGlyphRun.glyphIndices)) {
-                         CGAffineTransform finalTextTransform = CGAffineTransformConcat(textTransform, runningGlobalTransform);
-                         deviceContext->SetTransform(__CGAffineTransformToD2D_F(finalTextTransform));
-                         deviceContext->DrawGlyphRun(origin, &individualGlyphRun, state.fillBrush.Get(), DWRITE_MEASURING_MODE_NATURAL);
-
-                         // Uses glyphAdvances to move each glyph
-                         runningGlobalTransform = CGAffineTransformTranslate(runningGlobalTransform, glyphRun->glyphAdvances[i], 0);
-                     }
-                     return S_OK;
-                 }));
+            Factory()->CreateTransformedGeometry(pathGeometry.Get(), __CGAffineTransformToD2D_F(finalTextTransform), &transformedGeometry));
+    }
+
+    HRESULT ret = Draw(_kCGCoordinateModeDeviceSpace, nullptr, [&](CGContextRef context, ID2D1DeviceContext* deviceContext) {
```

Comment:
It was before but I ended up capturing **a lot** so changed it to let the compiler figure it out.  I also don't like [&] but it seems clearer to me here than a whole line of variables being captured